_posts/2019-11-01-snowplow-r117-biskupin/index.md

```diff
@@ -0,0 +1,209 @@
+```json
+ssl-config {
+  keyManager = {
+    stores = [
+      {type = "PKCS12", classpath = false, path = ${CERT_FILE}, password = "pass" }
+    ]
+  }
+}
+```
+​
+### Default redirect endpoint disabling
+Another simple, but important security improvement is a possibility to disable the default event submission redirect. This allows users to disable the default endpoint in favour of custom user-defined url. 
+For example, following configuration will only allow redirects for custom-defined `/com.acme/redirect-me` endpoint, whereas the default `/r/tp2` will not be available.
+```json
+enableDefaultRedirect = false
+paths {
+  "/com.acme/redirect-me" = "/r/tp2"
+}
+```
+Note, that for backwards-compatibility, the endpoint is still exposed by default.
+
+## Referer parser refreshment
+
+Referer parser is a library that contains a list of possible referer hosts. We keep maintenance of `0.3.x` branch with backporting new identified hosts into old `0.3.0` version. Several new hosts were identified and added to the [`referer.yml`](https://s3-eu-west-1.amazonaws.com/snowplow-hosted-assets/third-party/referer-parser/referers-latest.yml)
+
+## IPv6 anonymization
+
+This release adds support for anonymization of IPv6 addresses. `anonSegments` parameter is added to enrichment configuration. It describes how many segments should be anonymized.
+
+The schema for the configuration of the enrichment has been updated to version `1-0-1`:
+
+```json
+{
+	"schema": "iglu:com.snowplowanalytics.snowplow/anon_ip/jsonschema/1-0-1",
+	"data": {
+		"name": "anon_ip",
+		"vendor": "com.snowplowanalytics.snowplow",
+		"enabled": true,
+		"parameters": {
+			"anonOctets": 1,
+			"anonSegments": 1
+		}
+	}
+}
+```
+
+The IPv6 anonymization changes are backwards compatible. The `anonOctets` param will be read and useed also for IPv6 in case of missing `anonSegments`. To fully use IPv6 anonymization it's better to use the new schema (version `1-0-1`, which has `anonSegments` param). The new code can be used with `1-0-0` schema version also.
+
+## Additional event fingerprint hashing methods
+
+Credits for this contribution goes to [miike](https://github.com/miike).
+
+From this version, besides MD5, new hashing algorithms are supported:
+
+* SHA1
+* SHA256
+* SHA384
+* SHA512
+
+The schema for the configuration of the enrichment has been updated to version `1-0-1`.
+
+## Support to the spot market for core instances
+
+EmrEtlRunner now supports [EC2 spot instances][spot-instances], which can significantly reduce cost of EMR cluster by making sure the optimal instance is used.
+
+In order to enable spot instances, add a `core_instance_bid` setting to your `config.yml` file. This setting specifies a bid for an hour of EC2 spot instance in USD.
+
+```yaml
+aws:
+  emr:
+    jobflow:
+      core_instance_bid: 0.3
+```
+
+
+# Upgrading
```

Comment:
I think that we're missing the list of the new versions of components that have been upgraded.

Reply:
I just removed it, I thought you both told me that the list is not part of the standard release blog post sections. I'll add it back.

Reply:
R116 [blog post](https://snowplowanalytics.com/blog/2019/09/12/snowplow-r116-madara-rider/#upgrading) contains link to new SSC artifact on Bintray  I think that you should also mention the Docker image
R115 [blog post](https://snowplowanalytics.com/blog/2019/07/18/snowplow-r115-sigiriya-emretlrunner/#upgrading) contains link to new EER on Bintray 
R114 [blog post](https://snowplowanalytics.com/blog/2019/05/20/snowplow-r114-polonnaruwa-enrichments/#upgrading) contains links to new artifacts 

Basically in the `Upgrading` section you need to put the links to the new versions of artifacts as well as the changes in the configuration of the components if any

Reply:
I have added links to artifacts in the `Upgrading` section, also I have added the components that only contains snowplow-common-enrich bump.
Should `Spark Enrich` be pointing to `.jar` from hosted assets?

Reply:
From R114 [blog post](https://snowplowanalytics.com/blog/2019/05/20/snowplow-r114-polonnaruwa-enrichments/#upgrading) :

Finally, if you are a batch pipeline user, a new Spark Enrich can be used by setting the new version in your EmrEtlRunner configuration:
```
enrich:
  version:
    spark_enrich: 1.18.0 # WAS 1.17.0
```
or directly make use of the new Spark Enrich available at:

`s3://snowplow-hosted-assets/3-enrich/spark-enrich/snowplow-spark-enrich-1.18.0.jar`

Reply:
Thanks :+1: